Product/Production/Adapters/General/CONNECTAdminGUI/src/test/java/gov/hhs/fha/nhinc/admingui/services/impl/SHA1PasswordServiceTest.java

```diff
@@ -1,103 +1,110 @@
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
+import junit.framework.Assert;
+
+import org.junit.Test;
+
+/**
+ * The Class SHA1PasswordServiceTest.
+ */
+/**
+ * @author msw
+ *
+ */
+public class SHA1PasswordServiceTest {
+
+    /**
+     * Test.
+     *
+     * @throws PasswordServiceException the password service exception
+     * @throws NoSuchAlgorithmException the no such algorithm exception
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    @Test
+    public void test() throws PasswordServiceException, NoSuchAlgorithmException, IOException {
+        byte[] candidatePassword = "password".getBytes();
+        byte[] salt = "ABCD".getBytes();
+        byte[] passwordHash = "eFw9+D8egYfAGv1QjUMdVzI9dtvwiH3Amc6XlBoXZj03ebwzuQU8yoYzyLtz40JOn69a7P8zqtT7A6lEyIMBmw=="
+                .getBytes();
+        PasswordService service = getSHA1PasswordService();
+        Assert.assertTrue("Password should match",service.checkPassword(passwordHash, candidatePassword, salt));
+    }
+
+    /**
+     * Test that an incorrect password returns false.
+     *
+     * @throws PasswordServiceException the password service exception
+     * @throws NoSuchAlgorithmException the no such algorithm exception
+     * @throws IOException Signals that an I/O exception has occurred.
+     */
+    @Test
+    public void testNegative() throws PasswordServiceException, NoSuchAlgorithmException, IOException {
+        byte[] candidatePassword = "candidatePassword".getBytes();
+        byte[] salt = "ABCD".getBytes();
+        byte[] passwordHash = "nottherighthash".getBytes();
+        PasswordService service = getSHA1PasswordService();
+        Assert.assertFalse("Password doesn't match",service.checkPassword(passwordHash, candidatePassword, salt));
+    }
+
+    @Test
+    public void generateHash() throws IOException, PasswordServiceException {
+        String salt = "ABCD";// generateRandomSalt();
+        String password = "password";
+        String sha1 = new String(calculateHash(salt.getBytes(), password.getBytes()));
+        Assert.assertTrue("It should less than 100 characters", sha1.length()<=100);
+        Assert.assertEquals("eFw9+D8egYfAGv1QjUMdVzI9dtvwiH3Amc6XlBoXZj03ebwzuQU8yoYzyLtz40JOn69a7P8zqtT7A6lEyIMBmw==", sha1);
+    }
+
+    private byte[] calculateHash(byte[] salt, byte[] password) throws IOException, PasswordServiceException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        outputStream.write(salt);
+        outputStream.write(password);
+
+        PasswordService service = getSHA1PasswordService();
+        return service.calculateHash(outputStream.toByteArray());
+    }
+
+    /**
+     * Gets the SHA1 password service.
+     *
+     * @return the SHA1 password service
+     */
+    private PasswordService getSHA1PasswordService() {

```

Comment:
If so then he may want to rename the entire class to SHA2PasswordServiceTest or create a new SHA2 test class.